### PR TITLE
Refactor some build scripts

### LINKED ISSUE #745: Consolidate UFS build
**Description**
The UFS build currently requires a user to determine whether to build for atmosphere-only (ATMW) or coupled (S2SW). This is no longer necessary, as the coupled executable can be used for atmosphere-only runs. The two options build with different CCPP suite lists, so these lists need to be combined

**Requirements**
Removal of special switches for build_ufs and build_all.

**Acceptance Criteria (Definition of Done)**
A UFS executable that is usable for all global-workflow applications.

**Dependencies**
None

### LINKED ISSUE #751: Build DataAtmosphere configuration of the UFS-weather-model
The global-workflow currently build the "coupled" model.  It is noted that the coupled model has the ability to run in atmosphere-only mode.

This issue is to extend that to enable building the model in data atmosphere configuration in support of ocean DA capabilities.

Explore if the data-atmosphere configuration in the ufs-weather-model needs to be built differently.
If so, enable switches to allow building this configuration.

### PR DESCRIPTION
**Pull request by WalterKolczynski-NOAA**
**Description**
Refactors the build_ufs script to allow the build of any UFS app to support future expansion of global-workflow capability. The default should be sufficient for most users, as S2SWA can be used for ATM, ATMA, S2S, and S2SW as well. The new format of the command is now:
```
build_ufs.sh [-a UFS_app][-v]
  -a UFS_app:
    Specify the UFS application to build. The default if none is provided is S2SWA
  -v:
    Turn on verbose mode (BUILD_VERBOSE)
```

build_all.sh is similarly refactored to include the same options. Also, a new `-c build_config` option is added to specify an alternative list of programs to build.
```
build_all.sh [-a UFS_app][-c build_config][-h][-v]
  -a UFS_app:
    Build a specific UFS app instead of the default
  -c build_config:
    Selectively build based on the provided config instead of the default config
  -h:
    Print usage message and exit
  -v:
    Run all scripts in verbose mode
```

partial_build.sh is also updated to take the new `-c` option to specify a build config.
```
partial_build.sh [-c config_file][-h][-v]
  -c config_file:
    Selectively build based on the provided config. The default if none is specified is gfs_build.cfg
  -h:
    Print usage message and exit
  -v:
    Run in verbose mode
```

In addition to the above, build_all and partial_build had their indentation redone and their usage/help statement updated/added. Also, the build configuration file was renamed to from fv3gfs_build.cfg to gfs_build.cfg.

Fixes #745
Fixes #751

**Type of change**
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

**How Has This Been Tested?**
- [x] Clone and Build tests on Hera

**Checklist**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes

### CODE REVIEW

**Comment by JessicaMeixner-NOAA** [comment]
@WalterKolczynski-NOAA this is a really nice feature !  Have you thought about having optional inputs to add options to the ufs build such as debug mode ( -DDEBUG=ON), 32 bit builds (-D32BIT=ON), or other similar compile time options (-DMULTI_GASES=ON, -DMOVING_NEST=ON)?

**Comment by WalterKolczynski-NOAA** [comment]
Debug is definitely a future possibility (for all build scripts), but we also don't want to give too many options up front either.

**Comment by JessicaMeixner-NOAA** [comment]
The 32bit build is going to be a need fairly soon, so it seems smart to plan ahead for it while you're making these similar types of changes.

**Comment by WalterKolczynski-NOAA** [comment]
@JessicaMeixner-NOAA with the removal of the command line option to specify CCPP suites, are there different suites we need to build with if we pass ATMW instead of the default list?

**Comment by aerorahul** [comment]
@JessicaMeixner-NOAA @WalterKolczynski-NOAA 
I have mixed feelings about giving the user too much flexibility in configuring the model build options from inside the workflow.
The workflow does not support all the different combinations of the model.  If the user wants to do debugging of the model, they should be able to build the executable from the tag **outside** of the workflow and drop the executable in the `exec` folder, rather than calling the `build_all.sh` script and going through the hoops of the global-workflow.
My 2c.

**Comment by JessicaMeixner-NOAA** [comment]
@aerorahul   I understand not wanting to give too much flexibility within the workflow.  However, I disagree with building the model outside of the workflow with the debug option. I'd just hard code it and use the script to keep things consistent.  Using a completely different build script, etc introduces possibility for unintentional divergence in options, etc.  If debug is needed, it can be manually hard-coded and instead of being added as a user-defined option. 

@WalterKolczynski-NOAA given: 
```
APP="S2SWA"
CCPP_SUITES="FV3_GFS_v16,FV3_GFS_v16_RRTMGP,FV3_GFS_v16_ugwpv1,FV3_GFS_v17_p8,FV3_GFS_v16_coupled_nsstNoahmpUGWPv1,FV3_GFS_v16_coupled_p7_rrtmgp,FV3_GFS_v17_coupled_p8"
```

I'd remove the "coupled" suites for ATMW and have: 
```
APP="ATMW"
CCPP_SUITES="FV3_GFS_v16,FV3_GFS_v16_RRTMGP,FV3_GFS_v16_ugwpv1,FV3_GFS_v17_p8"
```

**Comment by WalterKolczynski-NOAA** [comment]
Okay, if ATMW uses the same suites, I don't see a need to change it.

I would love to trim the list of suites down though, if you can suggest ones from earlier in prototype testing we no longer need.

**Comment by JessicaMeixner-NOAA** [comment]
I'd propose: 
CCPP_SUITES="FV3_GFS_v16,FV3_GFS_v17_p8,FV3_GFS_v16_coupled_nsstNoahmpUGWPv1,FV3_GFS_v17_coupled_p8"
since we need the P8 suite files which are FV3_GFS_v17_coupled_p8/FV3_GFS_v17_p8, and then we need something without Thompson MP for DA, so I'd recommend FV3_GFS_v16_coupled_nsstNoahmpUGWPv1 which is the p7 suite. The physics group might have additional suite files or requirements for their testing//development.

File: sorc/build_all.sh [hunk 1]
```diff
@@ -1,4 +1,5 @@
-#!/bin/sh
+#! /usr/bin/env bash
+
 set +x
```
Comment (aerorahul) [inline sorc/build_all.sh:3]: Does `set -eu` work? If so, please add.

**Comment by JessicaMeixner-NOAA** [comment]
Looking at the latest [suite files](https://github.com/NOAA-EMC/fv3atm/tree/1e9c3da5e0cc9efcc434cb3a2ea08a9f521939d2/ccpp/suites), the following should be removed: 
FV3_GFS_v16_RRTMGP and FV3_GFS_v16_coupled_p7_rrtmgp 

Also, the README should probably also be updated with these new options when they're finalized.

**Review by aerorahul — APPROVED** [review]
looks good.

**Comment by aerorahul** [comment]
> @aerorahul I understand not wanting to give too much flexibility within the workflow. However, I disagree with building the model outside of the workflow with the debug option. I'd just hard code it and use the script to keep things consistent. Using a completely different build script, etc introduces possibility for unintentional divergence in options, etc. If debug is needed, it can be manually hard-coded and instead of being added as a user-defined option.
> 
> @WalterKolczynski-NOAA given:
> 
> ```
> APP="S2SWA"
> CCPP_SUITES="FV3_GFS_v16,FV3_GFS_v16_RRTMGP,FV3_GFS_v16_ugwpv1,FV3_GFS_v17_p8,FV3_GFS_v16_coupled_nsstNoahmpUGWPv1,FV3_GFS_v16_coupled_p7_rrtmgp,FV3_GFS_v17_coupled_p8"
> ```
> 
> I'd remove the "coupled" suites for ATMW and have:
> 
> ```
> APP="ATMW"
> CCPP_SUITES="FV3_GFS_v16,FV3_GFS_v16_RRTMGP,FV3_GFS_v16_ugwpv1,FV3_GFS_v17_p8"
> ```

Ok.  As `Debug` is a very special case, it is for the user to edit `build_ufs.sh` and add the appropriate `DEBUG` flag to the `cmake -DAPP=${APP} -DCCPP_SUITES=${CCPP_SUITES} ..` line.

Out of curiosity, how often does one "debug" the model in the workflow rather than in the model regression tests controlled environment? And how many users do we have that need this level of convenience.

**Comment by JessicaMeixner-NOAA** [comment]
@aerorahul to capture what we just discussed offline: I think it's fine to not put in the debug option and just keep the flexibility to specifying an APP. Debug is not used very often when trying to debug the model for the workflow.

**Review by JessicaMeixner-NOAA — APPROVED** [review]

**Review by binli2337 — APPROVED** [review]
Looks good. Thanks!

File: sorc/build_all.sh (hunk 1 above)
Comment (WalterKolczynski-NOAA) [reply sorc/build_all.sh:3]: Sorry, missed this before I merged. But adding `set -eu` to all scripts is part of my long-term plans (see Issue #397).

**Comment by WalterKolczynski-NOAA** [comment]
> Looking at the latest [suite files](https://github.com/NOAA-EMC/fv3atm/tree/1e9c3da5e0cc9efcc434cb3a2ea08a9f521939d2/ccpp/suites), the following should be removed: FV3_GFS_v16_RRTMGP and FV3_GFS_v16_coupled_p7_rrtmgp
> 
> Also, the README should probably also be updated with these new options when they're finalized.

Looks like #795 is going to remove the dead suites, so I left it for there.

**Pull request merged**